tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterBuilderTests.cs

```diff
@@ -59,10 +61,26 @@ public void CreateServerMonitorFactory_should_return_expected_result(int connect
 
             var serverSettings = result._settings();
         }
+
+        [Fact]
+        public void StreamFactoryWrappers_should_be_called_in_the_correct_order()
+        {
+            var subject = new ClusterBuilder();
+            var calls = new List<int>();
+            subject.RegisterStreamFactory(factory => { calls.Add(1); return factory; });
+            subject.RegisterStreamFactory(factory => { calls.Add(2); return factory; });
+            subject.RegisterStreamFactory(factory => { calls.Add(3); return factory; });
+
+            subject._streamFactoryWrapper()(Mock.Of<IStreamFactory>());
```

Comment:
I'm not sure what this is doing. Do we need it here?

Reply:
here, we are checking that added StreamFactory wrappers are being called in the same order as they were added in the code. Also, that a logic in each next wrapper compliments previous ones

Reply:
Yeah, that's on the next line where we verify the list. I'm talking specifically about this _streamFactoryWrapper method call...

Reply:
As `RegisterStreamFactory` only configures delegates wrappers (and do Not run them), we need to call(somehow) this delegate to check the calling order (Note: I created this test to check not only the ticket behavior).

Reply:
So, I use reflection to get the delegates wrappers and to call him directly.

Reply:
Ah, I see... This line is actually invoking the stream factory. My bad.